File: javascript/bot.js
const Discord = require('discord.js');
const client = new Discord.Client();

function clean(text) {
	if (typeof(text) === "string")
		return text.replace(/`/g, "`" + String.fromCharCode(8203)).replace(/@/g, "@" + String.fromCharCode(8203));
	else
		return text;
}
myid = "666317117154525185";
client.on('ready', () => {
	client.user.setStatus('idle');
	console.log(`We have logged in as ${client.user.tag}`);
	console.startlat = "global";
	setTimeout(() => {
		startlat = client.ws.ping;
	}, 0);
});

client.on('message', message => {
	if (!message.guild) return;
	if (message.content.startsWith('stopjs--')) {
		message.channel.send("bye");
		setTimeout(() => {
			client.destroy();
		}, 0);
	}
	if (message.content.startsWith('kys--')) {
		message.channel.send("Bye! ***-javascript***");
		setTimeout(() => {
			client.destroy();
		}, 0);
	}
	if (message.content.startsWith('kick--')) {
		const user = message.mentions.users.first();
		if (user) {
			const member = message.guild.member(user);
			if (member) {
				member
					.kick('Optional reason that will display in the audit logs')
					.then(() => {
						message.channel.send(`Successfully kicked ${user.tag}`);
					})
					.catch(err => {
						message.channel.send('I was unable to kick the member');
						console.error(err);
					});
			} else {
				message.channel.send("That user isn't in this guild!");
			}
		} else {
			message.channel.send("You didn't mention the user to kick!");
		}
	}
	if (message.content.startsWith('ban--')) {
		const user = message.mentions.users.first();
		if (user) {
			const member = message.guild.member(user);
			if (member) {
				member
					.ban({
						reason: 'They were bad!',
					})
					.then(() => {
						message.channel.send(`Successfully banned ${user.tag}`);
					})
					.catch(err => {
						message.channel.send('I was unable to ban the member');
						console.error(err);
					});
			} else {
				message.channel.send("That user isn't in this guild!");
			}
		} else {
			message.channel.send("You didn't mention the user to ban!");
		}
	}
	const args = message.content.split(" ").slice(1);

	if (message.content.startsWith("js--")) {
		if (message.author.id !== myid) return;
		try {
			const code = args.join(" ");
			let evaled = eval(code);

			if (typeof evaled !== "string")
				evaled = require("util").inspect(evaled);

			message.channel.send(clean(evaled), {
				code: "xl"
			});
		} catch (err) {
			message.channel.send(`\`ERROR\` \`\`\`xl\n${clean(err)}\n\`\`\``);
		}
	}
	if (message.content.startsWith('ping--')) {
		var h = client.users.cache.get(myid);
		const embed = {
			"title": "Pong",
			"description": "*javascript*",
			"color": 0x592630,
			"footer": {
				"icon_url": `${h.avatarURL()}`,
				"text": `created by @${h.username}#${h.discriminator} <${myid}>`
			},
			"thumbnail": {
				"url": "https://cdn.discordapp.com/attachments/676846050228568090/677256001837334578/alphabet-blocks-icon.png"
			},
			"fields": [{
					"name": "Latency Now",
					"value": `${client.ws.ping}ms`
				},
				{
					"name": "Latency At Startup",
					"value": `${startlat}ms`
				}
			]
		};
		message.channel.send({
			embed
		});
	}
});

client.login('BOT-TOKEN-HERE');

File: python/bot.py
import requests
import discord
import json
import aiohttp
import random
import sys as s
from discord.ext import commands
import logging
#creating the bot obj with the log level of error
logging.basicConfig(level=logging.ERROR, format="[\033[33;1mpyt\033[36;1mhon\033[m] [%(asctime)s] [\033[32;1m%(levelname)s]\033[m] %(name)s: %(message)s")
token = 'BOT-TOKEN-HERE'
client = commands.Bot(command_prefix='', owner_id=666317117154525185)
hid = 666317117154525185
#change hid and owner_id to your id

#the ready event, sets the status and prints in the console when the bot is logged in
@client.event
async def on_ready():
    print(f'We have logged in as {client.user} with a latency of {round(client.latency*1000,2)}ms')
    global startlat
    tmplat = str(round(client.latency*1000,0))
    startlat = f'{tmplat[0:len(tmplat)-2]}ms'
    await client.change_presence(activity=discord.Game(name='insert status here'), status=discord.Status('idle'))

#this is the on_message event, this is what tells the bot to respond to messages
@client.event
async def on_message(message):
    if message.content.startswith('ping--'):
        msg = discord.Embed(title="Pong",description='*python*', color=0x592630)
        msg.set_thumbnail(url=client.user.avatar_url)
        msg.set_footer(text=f'created by @{client.get_user(hid)} <{hid}>', icon_url=client.get_user(hid).avatar_url)
        tmplat = str(round(client.latency*1000,0))
        msg.add_field(name='Latency Now',value=f'{tmplat[0:len(tmplat)-2]}ms',inline=False)
        msg.add_field(name='Latency At Startup',value=startlat,inline=False)
        await message.channel.send(embed=msg)
    #displays latency when a message starting with ping-- is sent

    #this is where the owner-only commands are
    if message.author.id == hid:
        if message.content.startswith('kys--') or message.content.startswith('stoppy--'):
            message.content = 'jsk--shutdown'
        if message.content.startswith('print--'): #prints in the console
            msgcont=message.content[7:len(message.content)]
            if msgcont.startswith(' '):
                msgcont = msgcont[1:len(msgcont)]
            print(msgcont)
            await message.add_reaction('\U00002705')
            return
        if message.content.startswith('echo--'): #sends a message then deletes the original msg
            msgcont=message.content[6:len(message.content)]
            await message.channel.send(content=msgcont)
            await message.delete()
            return
    #hacky command renames
    if message.content != '':
        if message.content.startswith('jsk-- help'):
            message.content = message.content.replace('jsk-- help', 'help jsk')
        if message.content.startswith('jsk--help'):
            message.content = message.content.replace('jsk--help', 'help jsk')
        if message.content.startswith('jsk-- '):
            message.content = message.content.replace('jsk-- ', 'jsk')
        if message.content.startswith('jsk--'):
            message.content = message.content.replace('jsk--', 'jsk ')
        if message.content[len(message.content)-1] == ' ':
            message.content = message.content.replace('jsk', 'jsk ')
            message.content = message.content[0:len(message.content)-1]
            if message.content[len(message.content)-1] == ' ':
                message.content = message.content[0:len(message.content)-1]
        await client.process_commands(message)
#an error handler
@client.event
async def on_command_error(ctx, error):
    if hasattr(error, 'original'):
        error = error.original
    if isinstance(error, commands.CommandNotFound):
        return
    await ctx.send(str(error))
#more hacky command renames and loads jsk devtools
client.load_extension('jishaku')
client.remove_command("jsk shutdown")
jsk = client.get_command("jsk")
@client.command(name="shutdown", aliases=["logout"])
async def jsk_shutdown(self, ctx: commands.Context):
    await ctx.send("Bye! ***-python***")
    await ctx.bot.logout()
#runs the bot
client.run(token)
